narwhal/executor/src/notifier.rs

```diff
@@ -61,11 +69,9 @@ impl<State: ExecutionState + Send + Sync + 'static> Notifier<State> {
                     )
                     .await;
             }
-            // this is temporary
-            // we will get explicit signal from consensus on where is commit boundary in the future
-            if index.batch_index + 1
-                == index.consensus_output.certificate.header.payload.len() as u64
-            {
+
+            let round = index.consensus_output.certificate.round();
+            if self.committee.leader(round) == self.name {
```

Comment:
Actually it looks like we'll notify the callback on each batch execution for the corresponding certificate - so it won't execute once.

I would suggest getting this signal from the upstream component instead of making `notifier` inferring the commit boundary by using the leader round as the `executor` and the `consensus` are two different components. In this case I would suggest either:
* having a `bool` value in the `index` that would suggest that this `index` is a commit boundary
* consume from the `rx_notifier` channel a tuple of `Some((index, batch, commit_boundary))`

Reply:
Replied below - while I think bool approach would work in theory, I think passing entire commit from consensus to executor can unlock more possibilities in the future